client/lib/sites-list/list.js

```diff
@@ -402,13 +417,16 @@ SitesList.prototype.getSite = function( siteID ) {
 		return false;
 	}
 
-	return find( this.get(), function( site ) {
-		// We need to check `slug` before `domain` to grab the correct site on certain
-		// clashes between a domain redirect and a Jetpack site, as well as domains
-		// on subfolders, but we also need to look for the `domain` as a last resort
-		// to cover mapped domains for regular WP.com sites.
-		return site.ID === siteID || site.slug === siteID || site.domain === siteID || site.wpcom_url === siteID;
-	} );
+	const state = this.getState();
+	let siteFromState;
+
+	if ( Number.isInteger( siteID ) ) {
+		siteFromState = getSite( state, siteID );
+	} else {
+		siteFromState = getSiteBySlug( state, siteID ) || getSiteByDomain( state, siteID ) || getSiteByUrl( state, siteID );
+	}
+
+	return siteFromState && this.createSiteObject( siteFromState );
```

Comment:
Yes, there is ;). This is just a quick draft, we will need to remove duplicate attributes, selectedSite, etc.